eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -224,7 +225,12 @@ jobs:
       - template: ../variables/globals.yml
 
     strategy:
-      matrix: ${{ parameters.TestMatrix }}
+      matrix:
+        ${{ each matrixEntry in parameters.TestMatrix }}:
+          # Skip TestFromSource jobs for SDKType data
+          ${{ if not(and(eq(matrixEntry.value.TestFromSource, 'true'), eq(parameters.SDKType, 'data'))) }}:
```

Comment:
TestFromSource will be difficult to get working for track 1 libraries and I'm not sure it is worth the efforts so skipping that test entry.